core/src/main/scala/com/eed3si9n/jarjarabrams/Shader.scala

```diff
@@ -55,24 +63,26 @@ object Shader {
 
     val proc = new JJProcessor(jjrules, verbose, true, null)
 
-    /*
-    jarjar MisplacedClassProcessor class transforms byte[] to a class using org.objectweb.asm.ClassReader.getClassName
-    which always translates class names containing '.' into '/', regardless of OS platform.
-    We need to transform any windows file paths in order for jarjar to match them properly and not omit them.
-     */
-    val sanitizedMappings =
-      mappings.map(f => if (f._2.contains('\\')) (f._1, f._2.replace('\\', '/')) else f)
-    val shadedInputStreams = sanitizedMappings.flatMap { f =>
+    { (inputStream, mapping) =>
+      /*
+      jarjar MisplacedClassProcessor class transforms byte[] to a class using org.objectweb.asm.ClassReader.getClassName
+      which always translates class names containing '.' into '/', regardless of OS platform.
+      We need to transform any windows file paths in order for jarjar to match them properly and not omit them.
+       */
+      val sanitizedMapping = if (mapping.contains('\\')) mapping.replace('\\', '/') else mapping
       val entry = new EntryStruct
-      entry.data = readAllBytes(f._1)
-      entry.name = f._2
+      entry.data = readAllBytes(inputStream)
+      entry.name = sanitizedMapping
       entry.time = -1
       entry.skipTransform = false
-      if (proc.process(entry)) Some(new ByteArrayInputStream(entry.data) -> entry.name)
-      else None
+      val shadedInputStream =
+        if (proc.process(entry)) Some(new ByteArrayInputStream(entry.data) {
+          override def close(): Unit = inputStream.close()
+        } -> entry.name)
+        else None
+      val excludes = proc.getExcludes
+      shadedInputStream.filterNot(a => excludes.contains(a._2))
```

Comment:
I'm worried about this, is it ok to call that for every entry? cc @alexarchambault

Reply:
Can't `val excludes = proc.getExcludes` be moved along with `val proc = …` above?

Reply:
Did it